WaterdogPE/src/main/java/cn/stevei5mc/serverhelper/waterdogpe/commands/base/CommandBase.java

```diff
@@ -1,33 +1,114 @@
 package cn.stevei5mc.serverhelper.waterdogpe.commands.base;
 
-import cn.stevei5mc.serverhelper.common.utils.CommonUtils;
 import cn.stevei5mc.serverhelper.waterdogpe.ServerHelperMain;
 import dev.waterdog.waterdogpe.command.Command;
 import dev.waterdog.waterdogpe.command.CommandSender;
 import dev.waterdog.waterdogpe.command.CommandSettings;
+import org.cloudburstmc.protocol.bedrock.data.command.*;
 
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * 下列方法有修改的需要再进行重写，只需重写不在列表之内的方法即可正常使用。<br>
+ * The following methods need to be overridden only if modifications are required. Simply override methods not in this list for normal usage.
+ *
+ * <br><br> onExecute <br> buildCommandOverloads <br> addSubCommand <br> checkPermission <br>
+ */
 public abstract class CommandBase extends Command {
+
     protected ServerHelperMain main = ServerHelperMain.getInstance();
+    private final ArrayList<SubCommandBase> subCmdList = new ArrayList<>();
+    private final ConcurrentHashMap<String, Integer> subCmdMap = new ConcurrentHashMap<>();
 
+    /**
+     * @param name Command name
+     * @param description Command description
+     * @param permission Command permission
+     */
     public CommandBase(String name, String description, String permission) {
-        super(name,CommandSettings.builder().setDescription(description).setPermission(permission).build());
+        super(name.toLowerCase(), CommandSettings.builder().setDescription(description).setPermission(permission).build());
     }
 
+    /**
+     * @param name Command name
+     * @param description Command description
+     * @param permission Command permission
+     * @param aliases Command aliases
+     */
     public CommandBase(String name, String description, String permission, String[] aliases) {
-        super(name,CommandSettings.builder().setDescription(description).setPermission(permission).setAliases(aliases).build());
+        super(name.toLowerCase(), CommandSettings.builder().setDescription(description).setPermission(permission).setAliases(aliases).build());
     }
 
     @Override
     public boolean onExecute(CommandSender sender, String s, String[] args) {
```

Comment:
在 `WaterdogPE/src/main/java/cn/stevei5mc/serverhelper/waterdogpe/commands/base/CommandBase.java` 中实现了复杂的命令系统。建议添加更多的单元测试来确保命令解析和执行逻辑的正确性。

```suggestion
// 建议添加单元测试，覆盖以下场景：
// 1. 命令权限检查
// 2. 子命令解析和执行
// 3. 参数验证
// 4. 错误处理和提示信息